// lnbits/lib/wallet.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LNBitsWalletClass = void 0;
const axios_1 = require("axios");
class LNBitsWalletClass {
    constructor(params) {
        this.adminKey = '';
        this.invoiceReadKey = '';
        this.endpoint = 'https://lnbits.com';
        this.walletDetails = () => __awaiter(this, void 0, void 0, function* () {
            this.api.defaults.headers['X-Api-Key'] = this.adminKey;
            return yield this.api
                .get(`/wallet`)
                .then((res) => {
                return res.data;
            })
                .catch((err) => {
                throw err;
            });
        });
        this.createInvoice = (params = {
            amount: 0,
            memo: '',
            out: false,
        }) => __awaiter(this, void 0, void 0, function* () {
            this.api.defaults.headers['X-Api-Key'] = this.invoiceReadKey;
            return yield this.api
                .post(`/payments`, params)
                .then((res) => {
                return res.data;
            })
                .catch((err) => {
                throw err;
            });
        });
        this.payInvoice = (params = {
            bolt11: '',
            out: true,
        }) => __awaiter(this, void 0, void 0, function* () {
            this.api.defaults.headers['X-Api-Key'] = this.adminKey;
            return yield this.api
                .post(`/payments`, params)
                .then((res) => {
                return res.data;
            })
                .catch((err) => {
                throw err;
            });
        });
        this.checkInvoice = (params) => __awaiter(this, void 0, void 0, function* () {
            this.api.defaults.headers['X-Api-Key'] = this.invoiceReadKey;
            return yield this.api
                .get(`/payments/${params.payment_hash}`)
                .then((res) => {
                return res.data;
            })
                .catch((err) => {
                throw err;
            });
        });
        this.adminKey = params.adminKey;
        this.invoiceReadKey = params.invoiceReadKey;
        this.endpoint = params.endpoint || this.endpoint;
        this.api = axios_1.default.create({
            baseURL: `${this.endpoint}/api/v1`,
            headers: {
                'Content-Type': 'application/json',
            },
        });
    }
}
exports.LNBitsWalletClass = LNBitsWalletClass;
