// lnbits/lib/withdraw.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LNBitsWithdrawClass = void 0;
const axios_1 = require("axios");
class LNBitsWithdrawClass {
    constructor(params) {
        this.adminKey = '';
        this.invoiceReadKey = '';
        this.endpoint = 'https://lnbits.com';
        this.getLinks = (params) => __awaiter(this, void 0, void 0, function* () {
            this.api.defaults.headers['X-Api-Key'] = this.invoiceReadKey;
            let url = '/links';
            if (params === null || params === void 0 ? void 0 : params.withdraw_id) {
                url = `/links/${params === null || params === void 0 ? void 0 : params.withdraw_id}`;
            }
            return yield this.api
                .get(url)
                .then((res) => {
                return res.data;
            })
                .catch((err) => {
                throw err;
            });
        });
        this.createLink = (params) => __awaiter(this, void 0, void 0, function* () {
            this.api.defaults.headers['X-Api-Key'] = this.adminKey;
            return yield this.api
                .post(`/links`, params)
                .then((res) => {
                return res.data;
            })
                .catch((err) => {
                throw err;
            });
        });
        this.updateLink = (params) => __awaiter(this, void 0, void 0, function* () {
            this.api.defaults.headers['X-Api-Key'] = this.adminKey;
            return yield this.api
                .put(`/links/${params.withdraw_id}`, params)
                .then((res) => {
                return res.data;
            })
                .catch((err) => {
                throw err;
            });
        });
        this.deleteLink = (params) => __awaiter(this, void 0, void 0, function* () {
            this.api.defaults.headers['X-Api-Key'] = this.adminKey;
            return yield this.api
                .delete(`/links/${params.withdraw_id}`)
                .then(() => {
                return true;
            })
                .catch(() => {
                return false;
            });
        });
        this.adminKey = params.adminKey;
        this.invoiceReadKey = params.invoiceReadKey;
        this.endpoint = params.endpoint || this.endpoint;
        this.api = axios_1.default.create({
            baseURL: `${this.endpoint}/withdraw/api/v1`,
            headers: {
                'Content-Type': 'application/json',
            },
        });
    }
}
exports.LNBitsWithdrawClass = LNBitsWithdrawClass;
